### ISSUE TITLE
Monospace Image Font does not handle '-' range correctly

### ISSUE THREAD

**Issue by JeremyEastham** [godotengine/godot#82880]
### Godot version

4.2.dev5.mono

### System information

Godot v4.2.dev5.mono - Windows 10.0.19045 - Vulkan (Mobile) - integrated Intel(R) Iris(R) Xe Graphics (Intel Corporation; 30.0.100.9836) - 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz (8 Threads)

### Issue description

When importing an image as a monospace image font, the parsing of character ranges does not work properly. This is how each range is supposed to be parsed, according to [this page](https://docs.godotengine.org/en/stable/tutorials/ui/gui_using_fonts.html#bitmap-fonts) in the documentation:
> The Character Ranges option is an array that maps each position on the image (in tile coordinates, not pixels). The font atlas is traversed from left to right and top to bottom. Characters can be specified with decimal numbers (127), hexadecimal numbers (0x007f) or between single quotes ('~'). Ranges can be specified with a hyphen between characters.
> 
> For instance, 0-127 (or 0x0000-0x007f) denotes the full ASCII range. As another example, ' '-'~' is equivalent to 32-127 and denotes the range of printable (visible) ASCII characters.

The problem with the current algorithm is that it splits the range on dashes before each character is parsed as a codepoint:
https://github.com/godotengine/godot/blob/d31794c4a26e5e10fc30c34a1ae9722fd9f50123/editor/import/resource_importer_imagefont.cpp#L133-L148
https://github.com/godotengine/godot/blob/d31794c4a26e5e10fc30c34a1ae9722fd9f50123/editor/import/resource_importer_imagefont.cpp#L74-L90

### Steps to reproduce

1. Create a new 2D project
2. Create an image of a dash character and place it in the root of the project directory
3. Import this image as `Font Data (Monospace Image Font)`
4. Add a character range of `'-'`
5. Click `Reimport (*)`
6. Observe in the Output window that the following is printed:
```
editor/import/resource_importer_imagefont.cpp:136 - Invalid range: "'-'"
```
7. Create a label with the text `-`
8. Set the font to the dash image
9. Observe that the dash is not rendered correctly and that there is an error in the scene tree:

![image](https://github.com/godotengine/godot/assets/34139712/dc993e33-9533-4300-8e27-bc1ae70965a2)

### Minimal reproduction project

[Dash Bug Demo.zip](https://github.com/godotengine/godot/files/12823430/Dash.Bug.Demo.zip)

**Comment by JeremyEastham** [comment]
This would be a good first issue for someone. I'm considering fixing it myself, but I need to set up engine compilation.

**Comment by k0T0z** [comment]
@JeremyEastham can I help you with this?

**Comment by k0T0z** [comment]
@JeremyEastham Thank you. I think I have done it.

**Comment by k0T0z** [comment]
@JeremyEastham Can you check now? #82911

**Comment by JeremyEastham** [comment]
This doesn't appear to fix the issue completely. What about the range `'-'-'9'`, for example?

**Comment by k0T0z** [comment]
@JeremyEastham I added a whole new function and now all corner cases pass 👌.

**Comment by akien-mga** [comment]
Fixed by #88088.
